diff --git a/dbt/adapters/clickhouse/httpclient.py b/dbt/adapters/clickhouse/httpclient.py
index e809bca..001e5c6 100644
--- a/dbt/adapters/clickhouse/httpclient.py
+++ b/dbt/adapters/clickhouse/httpclient.py
@@ -26,7 +26,7 @@ class ChHttpClient(ChClientWrapper):
     def columns_in_query(self, sql: str, **kwargs) -> List[ClickHouseColumn]:
         try:
             query_result = self._client.query(
-                f"SELECT * FROM ( \n" f"{sql} \n" f") LIMIT 0",
+                f"SELECT * FROM ({sql}) WHERE FALSE LIMIT 0",
                 **kwargs,
             )
             return [
diff --git a/dbt/adapters/clickhouse/nativeclient.py b/dbt/adapters/clickhouse/nativeclient.py
index e2864cf..b5d8c72 100644
--- a/dbt/adapters/clickhouse/nativeclient.py
+++ b/dbt/adapters/clickhouse/nativeclient.py
@@ -35,7 +35,7 @@ class ChNativeClient(ChClientWrapper):
     def columns_in_query(self, sql: str, **kwargs) -> List[ClickHouseColumn]:
         try:
             _, columns = self._client.execute(
-                f"SELECT * FROM ( \n" f"{sql} \n" f") LIMIT 0",
+                f"SELECT * FROM ({sql}) WHERE FALSE LIMIT 0",
                 with_column_types=True,
             )
             return [ClickHouseColumn.create(column[0], column[1]) for column in columns]
